### DISCUSSION TITLE
extract tables , how to set the parameters to get the table?

### DISCUSSION THREAD

**Discussion by cobaltautomationdev** [jsvine/pdfplumber#1101] [Ask for help with specific PDFs]
cropped_page = page.within_bbox((22, 372, 592, 480))
then extract table in cropped_page region. I only want to extract the cropped_page. But I don't know how to set the table_settings.
![test](https://github.com/jsvine/pdfplumber/assets/145321686/07102e6d-ba71-42ad-aa88-727c01095971)

[04000000032611883.pdf](https://github.com/jsvine/pdfplumber/files/14401763/04000000032611883.pdf)

**Comment by jsvine** [comment]
Hi @cobaltautomationdev, and thanks for your interest in pdfplumber. To extract a table only from `cropped_page`, you can run `cropped_page.extract_table(table_settings)`.

To get a better sense of how the default settings would assess the extraction, you can run `cropped_page.to_image().debug_tablefinder()`. Depending on what you see, you may want to adjust the `horizontal_strategy` to `text`, or experiment with other settings.

**Reply by yonglee7015** [reply]
Thanks for your reply.Indeed I have tried what I can do,but still can't get a better way to extract the table. The extract page couldn't get a table. I tested Lines ,text,reacts and so on,but still couldn't get a table.

**Reply by jsvine** [reply]
Looking at the results of `page.to_image().debug_tablefinder()` with the default settings, it looks like the main issue is that the left and right borders aren't explicitly defined:

<img width="592" alt="Screenshot" src="https://github.com/jsvine/pdfplumber/assets/534702/c22ef7e0-7dc7-4d95-85b8-d7d1c612eb14">

A common strategy for dealing with that is to pass those positions explicitly using the `"explicit_vertical_lines": [...]` table setting, and deriving those positions from the other graphical elements on the page, such as the thick black rectangles that span the width of the table.

**Reply by yonglee7015** [reply]
actully I have cropped this part out (cropped_page = page.within_bbox((22, 372, 592, 480))) ,after cropped this part ,it couldn't find table  as it in a whole page.
